Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -418,14 +418,21 @@ class Connection : public ThreadSafeRefCounted<Connection, WTF::DestructionThrea
     struct WaitForMessageState;
     WaitForMessageState* m_waitingForMessage WTF_GUARDED_BY_LOCK(m_waitForMessageLock) { nullptr }; // NOLINT
 
-    class SyncMessageState;
+    class SyncMessageState : public ThreadSafeRefCounted<SyncMessageState> {
+    public:
+        virtual ~SyncMessageState() = default;
+    };
+    class SyncMessageStateImpl;
+    SyncMessageStateImpl& syncState();
 
     Lock m_syncReplyStateLock;
     bool m_shouldWaitForSyncReplies WTF_GUARDED_BY_LOCK(m_syncReplyStateLock) { true };
     bool m_shouldWaitForMessages WTF_GUARDED_BY_LOCK(m_waitForMessageLock) { true };
     struct PendingSyncReply;
     Vector<PendingSyncReply> m_pendingSyncReplies WTF_GUARDED_BY_LOCK(m_syncReplyStateLock);
 
+    Ref<SyncMessageState> m_syncState;
```

Comment:
Most likely this cannot be a refcounted, so you end up with SyncMessageState* and then you don't need the Impl rename in the .cpp.